src/Checkpointer.jl

```diff
@@ -46,11 +46,16 @@ function checkpoint_model_state(
 end
 
 """
-    restart_model_state!(sim::Interfacer.ComponentModelSimulation, t::Int; input_dir = "input")
+    restart_model_state!(sim::Interfacer.ComponentModelSimulation, comms_ctx::ClimaComms.AbstractCommsContext, t::Int; input_dir = "input")
 
 Sets the model state of a simulation from a HDF5 file from a given time, t (in seconds).
 """
-function restart_model_state!(sim::Interfacer.ComponentModelSimulation, t::Int; input_dir = "input")
+function restart_model_state!(
+    sim::Interfacer.ComponentModelSimulation,
+    comms_ctx::ClimaComms.AbstractCommsContext,
```

Comment:
This would be a breaking change right? Perhaps its effect is limited to the coupler so this doesn't affect any release schedules. LGTM.

Reply:
Yeah, there are no dependencies on this. Admittedly, this is what should have been in the original Checkpointer PR a few days ago.